### PR TITLE
[FEATURE] Add Rule generator

### PR DESCRIPTION
**Pull request by simonschaufi**
Resolves: #108

### CODE REVIEW

File: packages/fractor-rule-generator/composer.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
+    "name": "a9f/fractor-rule-generator",
+    "description": "Fractor rule generator",
+    "license": "MIT",
+    "type": "fractor-extension",
+    "authors": [
+        {
+            "name": "Andreas Wolf",
```
Comment (andreaswolf) [inline packages/fractor-rule-generator/composer.json:8]: I actually did not do anything here ;-)

File: packages/fractor-rule-generator/src/Factory/TemplateFactory.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace a9f\FractorRuleGenerator\Factory;
+
+class TemplateFactory
```
Comment (andreaswolf) [inline packages/fractor-rule-generator/src/Factory/TemplateFactory.php:7]: Could probably be final?

File: packages/fractor-rule-generator/src/Console/Command/GenerateRuleCommand.php [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+        $testCaseDirectoryPath = $this->resolveTestCaseDirectoryPath($generatedFilePaths);
+        $this->printSuccess($recipe->getFractorName(), $generatedFilePaths, $testCaseDirectoryPath);
+
+        return Command::SUCCESS;
+    }
+
+    private function askForTypo3Version(): Question
+    {
+        $whatTypo3Version = new Question('TYPO3-Version (i.e. 12.0): ');
```
Comment (andreaswolf) [inline packages/fractor-rule-generator/src/Console/Command/GenerateRuleCommand.php:103]: e.g. ("exempli gratia = for example"), not i.e. ("id est = that is")

There's various other places with that error as well.

File: packages/fractor-rule-generator/src/Console/Command/GenerateRuleCommand.php [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+            }
+        );
+
+        return $whatTypo3Version;
+    }
+
+    private function askForChangelogUrl(): Question
+    {
+        $whatIsTheUrlToChangelog = new Question(
+            'Url to changelog (i.e. https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/...) or "x" for none: '
```
Comment (andreaswolf) [inline packages/fractor-rule-generator/src/Console/Command/GenerateRuleCommand.php:123]: URL

**Review by andreaswolf — COMMENTED** [review]
Incomplete review, did not have a look at the inner workings yet.

File: packages/fractor-rule-generator/src/FileSystem/ConfigFilesystem.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+declare(strict_types=1);
+
+namespace a9f\FractorRuleGenerator\FileSystem;
+
+use a9f\Fractor\Exception\ShouldNotHappenException;
+use a9f\FractorRuleGenerator\Factory\TemplateFactory;
+use Nette\Utils\Strings;
+use Symfony\Component\Filesystem\Filesystem;
+
+final readonly class ConfigFilesystem
```
[the remark this answers is not in the document]
Comment (simonschaufi) [reply packages/fractor-rule-generator/src/FileSystem/ConfigFilesystem.php:12]: ask Sebastian. Copied the class from TYPO3 Rector

File: packages/fractor-rule-generator/composer.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    },
+    "config": {
+        "allow-plugins": {
+            "a9f/fractor-extension-installer": true
+        },
+        "sort-packages": true
+    },
+    "extra": {
+        "branch-alias": {
+            "dev-main": "0.3-dev"
```
[the remark this answers is not in the document]
Comment (simonschaufi) [reply packages/fractor-rule-generator/composer.json:46]: the other packages are also on `0.3-dev`.

File: packages/fractor-rule-generator/src/Contract/Typo3FractorTypeInterface.php [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace a9f\FractorRuleGenerator\Contract;
+
+interface Typo3FractorTypeInterface extends \Stringable
+{
+    public function getFolderName(): string;
+
+    public function getUseImports(): string;
+
+    public function getExtendsImplements(): string;
```
[the remark this answers is not in the document]
Comment (simonschaufi) [reply packages/fractor-rule-generator/src/Contract/Typo3FractorTypeInterface.php:13]: I thought about this as well but then we need some templating logic to only render the `implemens xxx` when the array is not empty which is just way more complex than just a string for the whole thing.

**Pull request merged**